### ISSUE TITLE
createdump's --full option can run out of memory

### ISSUE THREAD

**Issue by mikem8361** [dotnet/runtime#13003]
See issue: https://github.com/dotnet/coreclr/issues/25478

**Comment by felipepessoto** [comment]
@mikem8361, do you know why a simple web application (like the one from dotnet new mvc), generates a 12GB dump file when using -f?

What is the difference between -f and -h?

Thanks

**Comment by mikem8361** [comment]
I assume you are running 2.2 or less.  "-f" means "full" dump (all memory in the process include the module images) and -h is "withheap" which is all the necessary managed memory and state but not the module images. In 2.2 and less, createdump added all the uncommitted memory to the coredump which is unnecessary and made the dumps huge. This was changed/fixed in 3.x and greater. 

The goal with -h/"withheap" is to make the coredump as small as possible but still contain all the necessary managed state (GC heaps, managed threads, etc.).

**Comment by felipepessoto** [comment]
Do you if I can analyze a dump from a application running on Red Hat, using Ubuntu? "-f" would make any difference?

**Comment by mikem8361** [comment]
Yes, you should be able to use lldb/[SOS](https://github.com/dotnet/diagnostics/blob/master/documentation/installing-sos-instructions.md) or [dotnet-dump](https://github.com/dotnet/diagnostics/blob/master/documentation/dotnet-dump-instructions.md) to analyze the managed state (clrstack, dumpheap, etc.).

**Comment by mahamr** [comment]
Adding some context around what errors could pop-up when hitting this particular OOM issue.

As noted in the referenced [GitHub issue 25478](https://github.com/dotnet/coreclr/issues/25478), the stdout log for the .NET Core application could have messages like this:
```
Writing 33 thread entries to core file
Writing 1076 memory regions to core file
WriteData FAILED Cannot allocate memory
*** stack smashing detected ***: <unknown> terminated
```

Trying to open the dump in gdb produced output like this:
```
Cannot access memory at address 0x7f3e6ac93148
Cannot access memory at address 0x7f3e6ac93140
Failed to read a valid object file image from memory.
```

Neither lldb nor gdb were able to see any modules in the dumps, as well.
We're not 100% sure if the "stack smashing" error was related to the OOM issue or the original issue being troubleshooted, but wanted to call it out here just in case.
